code/datums/components/construction.dm

```diff
@@ -91,8 +91,9 @@
 				. = user.transferItemToLoc(I, parent)
 
 			// Using stacks
-			else if(istype(I, /obj/item/stack))
-				. = I.use_tool(parent, user, 0, volume=50, amount=current_step["amount"])
+			else
+				if(istype(I, /obj/item/stack))
+					. = I.use_tool(parent, user, 0, volume=50, amount=current_step["amount"])
```

Comment:
`else if()` is not a valid switch case nor is it evaluated as such. In this case there is no undesirable behavior, but it's still better to fix it.